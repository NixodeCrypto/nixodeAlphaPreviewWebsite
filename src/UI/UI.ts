import { css, Theme } from '@emotion/react';
import { rem } from 'polished';
import colorSwatches from '@/utils/colorSwatches'; // import directly to avoid circular dependency with utils/token

export const GlobalTheme: Theme = {
  space: {
    '-xl': rem(-64),
    '-lg': rem(-32),
    '-md': rem(-24),
    '-sm': rem(-16),
    '-xs': rem(-8),
    '-xss': rem(-4),
    xss: rem(4),
    xs: rem(8),
    sm: rem(16),
    md: rem(24),
    lg: rem(32),
    xl: rem(64),
  },
  fontSizes: {
    // Footer fs
    footer: rem(10),
    // Caption fs
    captionSm: rem(11),
    captionLg: rem(12),
    // Body fs
    bodySm: rem(14),
    bodyLg: rem(16),
    // H6 fs
    h6Sm: rem(12),
    h6: rem(14),
    // H5 fs
    h5Sm: rem(14),
    h5: rem(16),
    // H4 fs
    h4Sm: rem(16),
    h4: rem(20),
    // H3 fs
    h3Sm: rem(20),
    h3: rem(24),
    // H2 fs
    h2Sm: rem(24),
    h2: rem(34),
    // H1 fs
    h1Sm: rem(34),
    h1: rem(52),
  },
  colors: {
    // Brand Colors
    primary: colorSwatches('#FD9500'),
    secondary: colorSwatches('#82B5B2'),
    accent: colorSwatches('#DC5D3C'),
    // Base Colors
    red: colorSwatches('#f44336'),
    pink: colorSwatches('#e91e63'),
    purple: colorSwatches('#9c27b0'),
    deepPurple: colorSwatches('#673ab7'),
    indigo: colorSwatches('#3f51b5'),
    blue: colorSwatches('#2196f3'),
    lightBlue: colorSwatches('#03a9f4'),
    cyan: colorSwatches('#00bcd4'),
    teal: colorSwatches('#009688'),
    green: colorSwatches('#4caf50'),
    lightGreen: colorSwatches('#8bc34a'),
    lime: colorSwatches('#cddc39'),
    yellow: colorSwatches('#ffeb3b'),
    amber: colorSwatches('#ffc107'),
    orange: colorSwatches('#ff9800'),
    deepOrange: colorSwatches('#ff5722'),
    brown: colorSwatches('#795548'),
    grey: colorSwatches('#9e9e9e'),
    blueGrey: colorSwatches('#607d8b'),
  },
  fonts: {
    title: 'Inter, Roboto, sans-serif',
    text: 'Inter, Roboto, sans-serif',
  },
  fontWeights: {
    thin: 100,
    light: 300,
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700,
    extraBold: 800,
  },
  lineHeights: {},
  letterSpacings: {
    sm: rem(1.5),
  },
  sizes: {
    xss: rem(14),
    xs: rem(19),
    sm: rem(28),
    md: rem(37),
    lg: rem(46),
    xl: rem(64),
    xxl: rem(144),
    card: rem(215),
    cardXl: rem(225),
    maxLayoutSm: rem(320),
    maxLayoutMd: rem(400),
    img: rem(432),
    imgXl: rem(480),
    input: rem(280),
    inputXl: rem(480),
    max: '100%',
  },
  screenMaxWidths: {
    xs: rem(400),
    sm: rem(700),
    md: rem(1100),
  },
  breakpoints: {
    xss: '0px',
    xs: '320px',
    sm: '640px',
    md: '960px',
    lg: '1280px',
  },
  borders: {
    sm: `${rem(2)} solid`,
  },
  radii: {
    sm: rem(4),
    xl: rem(24),
  },
  shadows: {
    focus: '0 0 0 3px rgba(66,153,225,0.3)',
    reg: 'rgba(0,0,0,0.1) 0px 4px 12px',
  },
  zIndices: {
    navbar: 9999,
    popup: 6000,
  },
  transition: {
    standard: 'all 0.1s ease-in-out, width 0s, height 0s',
  },
};

export const GlobalReset = css`
  html {
    line-height: 1.15; /* 1 */
    -webkit-text-size-adjust: 100%; /* 2 */
    scroll-behavior: smooth;
  }
  body {
    margin: 0;
  }
  main {
    display: block;
  }
  h1 {
    font-size: 2em;
    margin: 0 0;
  }
  hr {
    box-sizing: content-box; /* 1 */
    height: 0; /* 1 */
    overflow: visible; /* 2 */
  }
  pre {
    font-family: monospace, monospace; /* 1 */
    font-size: 1em; /* 2 */
  }
  a {
    background-color: transparent;
  }
  abbr[title] {
    border-bottom: none; /* 1 */
    text-decoration: underline; /* 2 */
    text-decoration: underline dotted; /* 2 */
  }
  b,
  strong {
    font-weight: bolder;
  }
  code,
  kbd,
  samp {
    font-family: monospace, monospace; /* 1 */
    font-size: 1em; /* 2 */
  }
  small {
    font-size: 80%;
  }
  sub,
  sup {
    font-size: 75%;
    line-height: 0;
    position: relative;
    vertical-align: baseline;
  }
  sub {
    bottom: -0.25em;
  }
  sup {
    top: -0.5em;
  }
  img {
    border-style: none;
  }
  button,
  input,
  optgroup,
  select,
  textarea {
    font-family: inherit; /* 1 */
    font-size: 100%; /* 1 */
    line-height: 1.15; /* 1 */
    margin: 0; /* 2 */
  }
  button,
  input {
    /* 1 */
    overflow: visible;
  }
  button,
  select {
    /* 1 */
    text-transform: none;
  }
  button,
  [type='button'],
  [type='reset'],
  [type='submit'] {
    -webkit-appearance: button;
  }
  button::-moz-focus-inner,
  [type='button']::-moz-focus-inner,
  [type='reset']::-moz-focus-inner,
  [type='submit']::-moz-focus-inner {
    border-style: none;
    padding: 0;
  }
  button:-moz-focusring,
  [type='button']:-moz-focusring,
  [type='reset']:-moz-focusring,
  [type='submit']:-moz-focusring {
    outline: 1px dotted ButtonText;
  }
  fieldset {
    padding: 0.35em 0.75em 0.625em;
  }
  legend {
    box-sizing: border-box; /* 1 */
    color: inherit; /* 2 */
    display: table; /* 1 */
    max-width: 100%; /* 1 */
    padding: 0; /* 3 */
    white-space: normal; /* 1 */
  }
  progress {
    vertical-align: baseline;
  }
  textarea {
    overflow: auto;
  }
  [type='checkbox'],
  [type='radio'] {
    box-sizing: border-box; /* 1 */
    padding: 0; /* 2 */
  }
  [type='number']::-webkit-inner-spin-button,
  [type='number']::-webkit-outer-spin-button {
    height: auto;
  }
  [type='search'] {
    -webkit-appearance: textfield; /* 1 */
    outline-offset: -2px; /* 2 */
  }
  [type='search']::-webkit-search-decoration {
    -webkit-appearance: none;
  }
  ::-webkit-file-upload-button {
    -webkit-appearance: button; /* 1 */
    font: inherit; /* 2 */
  }
  details {
    display: block;
  }
  summary {
    display: list-item;
  }
  template {
    display: none;
  }
  [hidden] {
    display: none;
  }
`;
