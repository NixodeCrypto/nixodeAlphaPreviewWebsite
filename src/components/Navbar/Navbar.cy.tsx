import { mount } from '@cypress/react';
import Navbar from '.';

describe('components/Navbar', () => {
  it('works', () => {
    cy.viewport(1400, 900);
    mount(<Navbar />);
    cy.get('a').contains('Home');
  });
});
