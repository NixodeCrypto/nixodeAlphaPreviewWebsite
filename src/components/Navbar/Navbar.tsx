/* @jsxImportSource @emotion/react */
import { useEffect, useState } from 'react';
import styled from '@emotion/styled';
import { Menu } from 'react-feather';
import { useRouter } from 'next/router';

import { css } from '@emotion/react';
import { token, mq, parseBreakpoint } from '@/utils';
import Button from '@/components/Button';
import Flex from '@/components/Flex';
import Box from '@/components/Box';
import Link from '@/components/Link';

const Wrapper = styled.div`
  position: fixed;
  z-index: ${token.zIndices('navbar')};
  width: 100%;
  background: white;
`;

const NavbarRoot = styled.div`
  height: ${token.space('xl')};
  border-bottom: ${token.borders('sm')};
  border-bottom-color: ${token.colors('grey.50')};
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Space = styled.div`
  width: calc(100% - ${token.space('lg')});
  ${mq('md')} {
    width: calc(100% - ${token.space('xl')});
  }
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const BrandLogo = styled.img`
  width: ${token.sizes('xxl')};
`;

const MobileDrawer = styled.div`
  border-bottom: ${token.borders('sm')};
  border-bottom-color: ${token.colors('grey.50')};
  padding-left: ${token.space('sm')};
  padding-top: ${token.space('md')};
  background: white;
  position: fixed;
  width: 100%;
  display: flex;
  flex-direction: column;
  & > * {
    padding-bottom: ${token.space('lg')};
  }
`;

const ButtonContainer = styled.div`
  button:nth-of-type(1) {
    margin-bottom: ${token.space('xs')};
  }
  max-width: 20rem;
  width: calc(100% - ${token.sizes('sm')});
`;

const NavLink = styled.a`
  font-family: ${token.fonts('text')};
  color: black;
  text-decoration: none;
  font-weight: ${token.fontWeights('bold')};
`;

const Mobile = styled.div`
  display: block;
  ${mq('md')} {
    display: none;
  }
`;

const Desktop = styled.div<{ fullWidth?: boolean; rightAlign?: boolean }>`
  display: none;
  ${mq('md')} {
    width: ${(props) => (props.fullWidth ? '100%' : 'auto')};
    display: flex;
    flex-direction: row;
    justify-content: ${(props) => (props.rightAlign ? 'flex-end' : 'center')};
    align-items: center;
    &:not(:last-child) > * {
      margin-right: ${token.space('md')};
    }
    &:not(:first-child) > * {
      margin-left: ${token.space('md')};
    }
  }
`;

const Gutter = styled.div`
  width: 100%;
`;

const Navbar = () => {
  const router = useRouter();
  const [activeDrawer, setActiveDrawer] = useState(false);

  const handleResize = () => {
    if (window.innerWidth >= parseBreakpoint('md')) {
      setActiveDrawer(false);
    }
  };

  const handleOpenDrawer = () => {
    setActiveDrawer(!activeDrawer);
  };

  useEffect(() => {
    window.addEventListener('resize', handleResize);
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  });

  return (
    <Wrapper>
      <NavbarRoot>
        <Space>
          <Gutter>
            <NavLink href="/">
              <BrandLogo src="BrandLogo.svg" />
            </NavLink>
          </Gutter>
          <Mobile>
            <Button
              variant="text"
              color="grey"
              size="icon"
              onClick={handleOpenDrawer}
            >
              <Menu color="black" />
            </Button>
          </Mobile>
          <Desktop>
            <NavLink href="/">Home</NavLink>
            <NavLink href="/prices">Prices</NavLink>
            <NavLink href="/portfolio">Portfolio</NavLink>
            <NavLink href="/learn">Learn</NavLink>
          </Desktop>
          <Desktop fullWidth rightAlign>
            <NavLink href="/signin">Sign in</NavLink>
            <Button onClick={() => router.push('/signup')}>Sign up</Button>
          </Desktop>
        </Space>
      </NavbarRoot>
      {activeDrawer && (
        <MobileDrawer>
          <NavLink href="/">Home</NavLink>
          <NavLink href="/prices">Prices</NavLink>
          <NavLink href="/portfolio">Portfolio</NavLink>
          <NavLink href="/learn">Learn</NavLink>
          <ButtonContainer>
            <Button maxWidth size="lg">
              Get started
            </Button>
            <Button maxWidth size="lg" variant="outlined">
              Sign in
            </Button>
          </ButtonContainer>
        </MobileDrawer>
      )}
    </Wrapper>
  );
};

const NewNav = () => {
  const router = useRouter();
  const [activeDrawer, setActiveDrawer] = useState(false);

  const handleResize = () => {
    if (window.innerWidth >= parseBreakpoint('md')) {
      setActiveDrawer(false);
    }
  };

  const handleOpenDrawer = () => {
    setActiveDrawer(!activeDrawer);
  };

  useEffect(() => {
    window.addEventListener('resize', handleResize);
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  });

  return (
    <Box position="fixed" zIndex={9999} width="100%" background="white">
      <Flex
        justifyContent="center"
        alignItems="center"
        borderBottom="sm"
        borderColor="grey.50"
        height="xl"
      >
        <Flex
          width={{
            xss: `calc(100% - ${token.space('lg')})`,
            md: `calc(100% - ${token.space('xl')})`,
          }}
          justifyContent="space-between"
          alignItems="center"
        >
          <Box width="100%">
            <Link href="/">
              <BrandLogo src="BrandLogo.svg" />
            </Link>
          </Box>
          <Box display={{ xss: 'block', md: 'none' }}>
            <Button
              variant="text"
              color="grey"
              size="icon"
              onClick={handleOpenDrawer}
            >
              <Menu color="black" />
            </Button>
          </Box>
          <Flex
            display={{ xss: 'none', md: 'flex' }}
            flexDirection="row"
            justifyContent="center"
            alignItems="center"
            horizontalGap={token.space('md')}
          >
            <Link href="/">Home</Link>
            <Link href="/prices">Prices</Link>
            <Link href="/portfolio">Portfolio</Link>
            <Link href="/learn">Learn</Link>
          </Flex>

          <Flex
            display={{ xss: 'none', md: 'flex' }}
            width="100%"
            flexDirection="row"
            justifyContent="flex-end"
            alignItems="center"
            horizontalGap={token.space('md')}
          >
            <Link href="/signin">Sign in</Link>
            <Button onClick={() => router.push('/signup')}>Sign up</Button>
          </Flex>
        </Flex>
      </Flex>
      {activeDrawer && (
        <Flex
          borderBottom="sm"
          borderBottomColor="grey.50"
          pl="sm"
          pt="md"
          background="white"
          position="fixed"
          width="100%"
          flexDirection="column"
          css={css`
            & > * {
              padding-bottom: ${token.space('lg')};
            }
          `}
        >
          <Link href="/">Home</Link>
          <Link href="/prices">Prices</Link>
          <Link href="/portfolio">Portfolio</Link>
          <Link href="/learn">Learn</Link>
          <Box
            width={`calc(100% - ${token.sizes('sm')})`}
            maxWidth="20rem"
            css={css`
              button:nth-of-type(1) {
                margin-bottom: ${token.space('xs')};
              }
            `}
          >
            <Button maxWidth size="lg">
              Get started
            </Button>
            <Button maxWidth size="lg" variant="outlined">
              Sign in
            </Button>
          </Box>
        </Flex>
      )}
    </Box>
  );
};
export default Navbar;
