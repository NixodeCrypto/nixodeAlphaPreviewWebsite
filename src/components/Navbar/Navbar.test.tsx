import { ReactWrapper, mount } from 'enzyme';
import { GlobalTheme } from '@/UI';
import Navbar from './Navbar';

let wrapper: ReactWrapper;

beforeAll(() => {
  wrapper = mount(<Navbar />);
});

describe('components/Navbar', () => {
  it('matches screenshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('zIndex', () => {
    expect(wrapper).toHaveStyleRule(
      'z-index',
      GlobalTheme.zIndices.navbar.toString(),
    );
  });
  it('correct links', () => {
    expect(wrapper.text().includes('Home'));
    expect(wrapper.text().includes('Prices'));
    expect(wrapper.text().includes('Portfolio'));
    expect(wrapper.text().includes('Learn'));
    expect(wrapper.text().includes('Sign in'));
    expect(wrapper.text().includes('Sign up'));
  });
});
