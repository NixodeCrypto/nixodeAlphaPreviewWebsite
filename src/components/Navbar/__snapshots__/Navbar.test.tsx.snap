// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar matches screenshot 1`] = `ReactWrapper {}`;
